packages/dd-trace/test/appsec/index.fastify.plugin.spec.js

```diff
@@ -264,6 +265,91 @@ withVersions('fastify', 'fastify', version => {
   })
 })
 
+describe('Api Security - Fastify', () => {
+  withVersions('fastify', 'fastify', version => {
+    let config, server, axios
+
+    before(() => {
+      return agent.load(['fastify', 'http'], { client: false })
+    })
+
+    before((done) => {
+      const fastify = require(`../../../../versions/fastify@${version}`).get()
+
+      const app = fastify()
+
+      app.post('/send', (request, reply) => {
+        reply.send({ sendResKey: 'sendResValue' })
+      })
+
+      app.post('/return', async (request, reply) => {
+        return { returnResKey: 'returnResValue' }
+      })
+
+      getPort().then((port) => {
+        app.listen({ port }, () => {
+          axios = Axios.create({ baseURL: `http://localhost:${port}` })
+          done()
+        })
+        server = app.server
+      })
+    })
+
+    after(() => {
+      server.close()
+      return agent.close({ ritmReset: false })
+    })
+
+    beforeEach(() => {
+      config = new Config({
+        appsec: {
+          enabled: true,
+          rules: path.join(__dirname, 'api_security_rules.json'),
+          apiSecurity: {
+            enabled: true,
+            sampleDelay: 10
+          }
+        }
+      })
+      appsec.enable(config)
+    })
+
+    afterEach(() => {
+      appsec.disable()
+    })
+
+    function formatSchema (body) {
+      return zlib.gzipSync(JSON.stringify(body)).toString('base64')
+    }
+
+    it('should get the response body schema with reply.send', async () => {
+      const expectedResponseBodySchema = formatSchema([{ sendResKey: [8] }])
+      const res = await axios.post('/send', { key: 'value' })
+
+      await agent.assertSomeTraces((traces) => {
+        const span = traces[0][0]
+        assert.equal(span.meta['_dd.appsec.s.res.body'], expectedResponseBodySchema)
+      })
```

Comment:
You could use the new `assertFirstTraceSpan` helper :)

The same in the spot below.

```suggestion
      await agent.assertFirstTraceSpan({
        meta: {
          '_dd.appsec.s.res.body': expectedResponseBodySchema
        }
      })
```

Reply:
Good to know. Done